File: app/src/main/java/com/sumian/hw/improve/widget/TabIndicatorItemView.java
package com.sumian.hw.improve.widget;

import android.content.Context;
import android.support.annotation.DrawableRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.text.TextUtils;
import android.util.AttributeSet;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import com.sumian.sd.R;

import butterknife.ButterKnife;

/**
 * Created by sm
 * on 2018/3/6.
 * desc:
 */

public class TabIndicatorItemView extends FrameLayout implements View.OnClickListener {

    TextView mTvTabText;
    View mIvTabDot;
    ImageView mIvCalendar;

    private OnSelectTabCallback mOnSelectTabCallback;

    public TabIndicatorItemView(@NonNull Context context) {
        this(context, null);
    }

    public TabIndicatorItemView(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public TabIndicatorItemView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setOnClickListener(this);
        initView(context);
    }

    private void initView(Context context) {
        View inflate = inflate(context, R.layout.hw_lay_tab_dot, this);
        ButterKnife.bind(inflate);
        mTvTabText = inflate.findViewById(R.id.tv_tab_text);
        mIvTabDot = inflate.findViewById(R.id.v_tab_dot);
        mIvCalendar = inflate.findViewById(R.id.iv_calendar);

        mTvTabText.setVisibility(GONE);
        mIvCalendar.setVisibility(GONE);
        mIvTabDot.setVisibility(GONE);
    }

    public void setOnSelectTabCallback(OnSelectTabCallback onSelectTabCallback) {
        mOnSelectTabCallback = onSelectTabCallback;
    }

    @Override
    public void onClick(View v) {
        if (mOnSelectTabCallback != null) {
            mTvTabText.setActivated(true);
            mOnSelectTabCallback.onSelect(v, mTvTabText.isActivated());
        }
    }

    public void unSelect() {
        mTvTabText.setActivated(false);
    }

    public void select() {
        mTvTabText.setActivated(true);
    }

    public void setIndicatorText(String text) {
        mTvTabText.setText(text);
        mTvTabText.setVisibility(!TextUtils.isEmpty(text) ? VISIBLE : GONE);
    }

    public void setCalendarIcon(@DrawableRes int drawable) {
        mIvCalendar.setImageResource(drawable);
        mIvCalendar.setVisibility(drawable > -1 ? VISIBLE : GONE);
    }

    public void showDot() {
        mIvTabDot.setVisibility(VISIBLE);
    }

    public void hideDot() {
        mIvTabDot.setVisibility(GONE);
    }


    public interface OnSelectTabCallback {

        void onSelect(View v, boolean isSelect);
    }

}

File: app/src/main/java/com/sumian/hw/widget/TabIndicatorItemView.kt
package com.sumian.hw.widget

import android.annotation.SuppressLint
import android.content.Context
import android.support.annotation.DrawableRes
import android.text.TextUtils
import android.util.AttributeSet
import android.view.View
import android.widget.FrameLayout
import com.sumian.common.widget.voice.IVisible
import com.sumian.sd.R
import kotlinx.android.synthetic.main.hw_lay_tab_dot.view.*

/**
 * Created by sm
 * on 2018/3/6.
 * desc:
 */

class TabIndicatorItemView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0)
    : FrameLayout(context, attrs, defStyleAttr), View.OnClickListener, IVisible {


    private var mOnSelectTabCallback: OnSelectTabCallback? = null

    init {
        setOnClickListener(this)
        initView(context)
    }

    private fun initView(context: Context) {
        setPadding(0, resources.getDimensionPixelSize(R.dimen.space_32), 0, resources.getDimensionPixelSize(R.dimen.space_14))
        View.inflate(context, R.layout.hw_lay_tab_dot, this)
        tv_tab_text.visibility = View.GONE
        iv_calendar.visibility = View.GONE
        v_tab_dot.visibility = View.GONE
    }

    fun setOnSelectTabCallback(onSelectTabCallback: OnSelectTabCallback) {
        mOnSelectTabCallback = onSelectTabCallback
    }

    override fun onClick(v: View) {
        if (mOnSelectTabCallback != null) {
            tv_tab_text.isActivated = true
            mOnSelectTabCallback!!.onSelect(v, tv_tab_text.isActivated)
        }
    }

    fun unSelect() {
        tv_tab_text.isActivated = false
    }

    fun select() {
        tv_tab_text.isActivated = true
    }

    fun setIndicatorText(text: String) {
        tv_tab_text.text = text
        tv_tab_text.visibility = if (!TextUtils.isEmpty(text)) View.VISIBLE else View.GONE
    }

    @SuppressLint("ResourceType")
    fun setCalendarIcon(@DrawableRes drawable: Int) {
        iv_calendar.setImageResource(drawable)
        iv_calendar.visibility = if (drawable > -1) View.VISIBLE else View.GONE
    }

    fun showDot() {
        v_tab_dot.visibility = View.VISIBLE
    }

    fun hideDot() {
        v_tab_dot.visibility = View.GONE
    }

    override fun show() {
        visibility = View.VISIBLE
    }

    override fun hide() {
        visibility = View.GONE
    }


    interface OnSelectTabCallback {

        fun onSelect(v: View, isSelect: Boolean)

    }

}
